code_challenges/advent_of_code/security_through_obscurity.py

```diff
@@ -0,0 +1,102 @@
+def get_room_names():
+    with open("./code_challenges/advent_of_code/room_name_input") as room_names_input:
+        room_names = room_names_input.read().split("\n")
+    return room_names
+
+
+def get_incrypted_room_name():
+    room_names = get_room_names()
+    room_names = [room.split("-") for room in room_names]
+    rooms = {}
+
+    for room in room_names:
+        letter_count = {}
+        for i in range(len(room) - 1):
+            for character in room[i]:
+                letter_count[character] = letter_count.get(character, 0) + 1
+        rooms[room[-1]] = letter_count
```

Comment:
This dictionary isn't very useful. Instead create a list of tuples `[(num, code, counts ), ....]`